local-cli/runIOS/runIOS.js

```diff
@@ -198,19 +198,22 @@ function buildProject(xcodeProject, udid, scheme, configuration = 'Debug', launc
         console.log(data.toString());
       }
     });
+    let error;
     buildProcess.stderr.on('data', function(data) {
-      console.error(data.toString());
+      error = data.toString();
+      console.error(error);
     });
     buildProcess.on('close', function(code) {
       if (xcpretty) {
         xcpretty.stdin.end();
       }
+      if (error) {
+        process.exit(1);
+      }
       //FULL_PRODUCT_NAME is the actual file name of the app, which actually comes from the Product Name in the build config, which does not necessary match a scheme name,  example output line: export FULL_PRODUCT_NAME="Super App Dev.app"
       let productNameMatch = /export FULL_PRODUCT_NAME="?(.+).app"?$/m.exec(buildOutput);
-      if (productNameMatch && productNameMatch.length && productNameMatch.length > 1) {
-        return resolve(productNameMatch[1]);//0 is the full match, 1 is the app name
-      }
-      return buildProcess.error ? reject(buildProcess.error) : resolve();
+      //[0] of productNameMatch is the full match, [1] is the app name
+      resolve(productNameMatch && productNameMatch.length && productNameMatch.length > 1 && productNameMatch[1] || undefined);
```

Comment:
I think I'd prefer a ternary expression here to make it easier to understand.

Reply:
Changed.